logging/structured/src/main/scala/tofu/logging/location/LocationLogging.scala

```diff
@@ -0,0 +1,110 @@
+    writeLocation(Warn, location, message, values: _*)
+  def error(message: String, location: Location, values: LoggedValue*): F[Unit] =
+    writeLocation(Error, location, message, values: _*)
+
+  def traceWithMarker(message: String, location: Location, marker: Marker, values: LoggedValue*): F[Unit] =
+    writeMarker(Trace, message, marker, location, values: _*)
+  def debugWithMarker(message: String, location: Location, marker: Marker, values: LoggedValue*): F[Unit] =
+    writeMarker(Debug, message, marker, location, values: _*)
+  def infoWithMarker(message: String, location: Location, marker: Marker, values: LoggedValue*): F[Unit]  =
+    writeMarker(Info, message, marker, location, values: _*)
+  def warnWithMarker(message: String, location: Location, marker: Marker, values: LoggedValue*): F[Unit]  =
+    writeMarker(Warn, message, marker, location, values: _*)
+  def errorWithMarker(message: String, location: Location, marker: Marker, values: LoggedValue*): F[Unit] =
+    writeMarker(Error, message, marker, location, values: _*)
+
+  def traceCause(message: String, location: Location, cause: Throwable, values: LoggedValue*): F[Unit] =
+    writeCause(Trace, location, message, cause, values: _*)
+  def debugCause(message: String, location: Location, cause: Throwable, values: LoggedValue*): F[Unit] =
+    writeCause(Debug, location, message, cause, values: _*)
+  def infoCause(message: String, location: Location, cause: Throwable, values: LoggedValue*): F[Unit]  =
+    writeCause(Info, location, message, cause, values: _*)
+  def warnCause(message: String, location: Location, cause: Throwable, values: LoggedValue*): F[Unit]  =
+    writeCause(Warn, location, message, cause, values: _*)
+  def errorCause(message: String, location: Location, cause: Throwable, values: LoggedValue*): F[Unit] =
+    writeCause(Error, location, message, cause, values: _*)
+
+}
+
+object LocationLogging {
+  def sync[I[_]: Sync, F[_]: Sync]: I[LocationLogging[F]] =
+    Sync[I].delay(LoggerFactory.getILoggerFactory).map { factory =>
+      new LocationLogging[F] {
+        override def writeLocation(level: Logging.Level, location: Location, message: String, values: LoggedValue*)
+            : F[Unit] =
+          for {
+            logger <- Sync[F].delay(factory.getLogger(location.loggerName)) //cached inside factory
+            logging = new SyncLogging[F](logger)
+            _      <- logging.write(level, message, values: _*)
+          } yield ()
+      }
+    }
+
+  def withContext[I[_]: Sync, F[_]: Sync](implicit ctx: LoggableContext[F]): I[LocationLogging[F]] = {
+    import ctx.loggable
+
+    Sync[I].delay(LoggerFactory.getILoggerFactory).map { factory =>
+      new LocationLogging[F] {
+        override def writeLocation(level: Logging.Level, location: Location, message: String, values: LoggedValue*)
+            : F[Unit] =
+          for {
+            logger <- Sync[F].delay(factory.getLogger(location.loggerName)) //cached inside factory
```

Comment:
This comment is excessive and misleaading

Reply:
Still I dont understand the whole point of `I` outside if we acquire the logger every time